pkg/controller/remotekubernetescluster/sync_finalizer.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 ScyllaDB.
+
+package remotekubernetescluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/helpers/slices"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+func (rkcc *Controller) syncFinalizer(ctx context.Context, rkc *scyllav1alpha1.RemoteKubernetesCluster) ([]metav1.Condition, error) {
```

Comment:
I recall your previous PRs in the enterprise repo having a separate `protection` controller in a dedicated file subtree - why the change?